### ISSUE TITLE
Questions about maintaining this repo

### ISSUE THREAD

**Issue by Tom-Newton** [Ackuq/spark-pit#41]
First, thank you for building this and open sourcing it. My current estimate is that the early stop sort merge provides a 25X speed up on one of our workloads compared to the union plus window function approach we were doing before. This makes make extremely happy!

This is kind of a strange github issue because I just want to ask some questions. I'm interested in potentially contributing to this and I don't want to just start bombarding you with stuff. 
1. If I make PRs would they get reviewed and merged? Likely PRs:
    1. Upgrade to Spark 3.5.0
    1. Switch to using `SparkSession` instead of `SQLContext` which is deprecated. 
    1. Some little edgecase bug fixes. 
1. I'm confused why there is both `doExecute` and `doProduce`, since both seem to contain very similar logic. It seems like `doExecute` only gets used when joining more than 2 datframes or if the input dataframes have >= 98 columns (this part confuses me even more). Probably I'm just missing some context but I think its quite likely that all the little bugs I found would not have happened if there was a single implementation. I would be very interested to know why its like this.

**Comment by Ackuq** [comment]
Hi Tom! I am very happy that this project has been found valuable to you 😁

Regarding maintenance of this repo I probably first need to give some context. This project was a part of my master thesis I did in the spring of 2022 (see here if you are interested: http://www.diva-portal.org/smash/get/diva2:1695672/FULLTEXT01.pdf). Since then I have not kept up with updating and maintaining this project. Will add this to the README. Furthermore, I have not really touched Spark since I stopped working on this project if I am going to be honest.

With that said, I believe the best way to go about the changes you are proposing and further development would probably be to do a fork of this repository. 

I’ll try answer 2) in my best capacity, The difference between these two is that `doProduce` is used when Whole-Stage Codegen is enabled. It is an optimization technique that can be read into more detail here: https://jaceklaskowski.gitbooks.io/mastering-spark-sql/content/spark-sql-whole-stage-codegen.html.
`doExecute` is used when Whole-Stage Codegen either cannot be used or is disabled. I believe that the Codegen is being disabled because you are reaching the max number of output fields, which by default is 100, see: https://jaceklaskowski.gitbooks.io/mastering-spark-sql/content/spark-sql-properties.html#spark.sql.codegen.maxFields. I wish I could go into more detail but my Spark knowledge is a bit too rusty.

Again, thanks for showing interest in this project! Amazing that it has been of value to you 😊
